src/components/views/messages/ReactionsRow.js

```diff
@@ -125,10 +136,23 @@ export default class ReactionsRow extends React.PureComponent {
                 reactionEvents={events}
                 myReactionEvent={myReactionEvent}
             />;
-        });
+        }).filter(item => !!item);
+
+        let showAllLink;
+        if (items.length > MAX_ITEMS_WHEN_LIMITED && !showAll) {
```

Comment:
something that's always bothered me is how read receipts say "+1". Although we don't have a count here, the same problem applies: it doesn't make much sense to hide 1 element with an element of the same size. Instead, we might want to consider truncating at showing 9 reactions and on the 10th only showing 8 plus the "show all" button.

Reply:
Hmm, I defer to @nadonomy on this.

Reply:
That sounds great to me from a UX perspective, but I'll defer to other minds on whether it's worth adding that logic now or later.

Reply:
Tweaked in https://github.com/matrix-org/matrix-react-sdk/pull/3152/commits/04398b7853d9fcf3ad38376bac61d533943231e7